src/compiler/checker.ts

```diff
@@ -4024,6 +4024,10 @@ namespace ts {
 
             function determineIfDeclarationIsVisible() {
                 switch (node.kind) {
+                    case SyntaxKind.JSDocTypedefTag:

```

Comment:
Need to handle SyntaxKind.CallbackTag too

Reply:
https://github.com/Microsoft/TypeScript/pull/23947 isn't merged yet